### ISSUE TITLE
upper-lower rule incorrectly flags situations where case matters (despite case-insensitive mode)

### ISSUE THREAD

**Issue by jgerner-sfmc** [tsqllint/tsqllint#90]
Usage of UPPER/LOWER is redundant when dealing with columns in a case-insensitive DB, but other non-column operations may need to use UPPER/LOWER.

Simplified example that would be flagged (incorrectly):
```
SELECT HASHBYTES('SHA1', 'foo');
SELECT HASHBYTES('SHA1', UPPER('foo'));
```

The two values differ: HASHBYTES operates case-sensitive, despite the DB case mode.

**Comment by nathan-boyd** [comment]
This rule should be limited to only flag comparisons in select statements.
```
SELECT * FROM Foo WHERE Status = lower(@Status)
```
